lib/ui/pages/score/score_page.dart

```diff
@@ -45,6 +45,93 @@ class _ScoreViewerPageState extends State<ScoreViewerPage> with TickerProviderSt
   int _currentTabIndex = 0;
   bool isLoading = true;
 
+  Widget get _summaryTile {
+    final titleWidget = _buildTile(sprintf("%s %d/%d", [
+      R.current.creditSummary,
+      courseScoreCredit.getTotalCourseCredit(),
+      courseScoreCredit.graduationInformation.lowCredit,
+    ]));
+
+    final widgetList = [
+      _buildType(constCourseType[0], R.current.compulsoryCompulsory),
+      _buildType(constCourseType[1], R.current.revisedCommonCompulsory),
+      _buildType(constCourseType[2], R.current.jointElective),
+      _buildType(constCourseType[3], R.current.compulsoryProfessional),
+      _buildType(constCourseType[4], R.current.compulsoryMajorRevision),
+      _buildType(constCourseType[5], R.current.professionalElectives),
+    ];
+
+    return AppExpansionTile(
+      title: titleWidget,
+      initiallyExpanded: appExpansionInitiallyExpanded,
+      children: widgetList,
+    );
+  }
+
+  Widget get _generalLessonItemTile {
+    final generalLesson = courseScoreCredit.getGeneralLesson();
+    final List<Widget> widgetList = [];
+    int selectCredit = 0;
+    int coreCredit = 0;
+
+    for (final courseScoreInfoList in generalLesson.values) {
+      for (final course in courseScoreInfoList) {
+        if (course.isCoreGeneralLesson) {
+          coreCredit += course.credit.toInt();
+        } else {
+          selectCredit += course.credit.toInt();
+        }
+
+        final courseItemWidget = _buildOneLineCourse(course.name, course.openClass);
+        widgetList.add(courseItemWidget);
+      }
+    }
+
+    final titleWidget = _buildTile(sprintf("%s \n %s:%d %s:%d", [
+      R.current.generalLessonSummary,
+      R.current.takeCore,
+      coreCredit,
+      R.current.takeSelect,
+      selectCredit,
+    ]));
+
+    return AppExpansionTile(
+      title: titleWidget,
+      initiallyExpanded: appExpansionInitiallyExpanded,
+      children: widgetList,
+    );
+  }
+
+  Widget get _otherDepartmentItemTile {
+    final department = LocalStorage.instance.getGraduationInformation().selectDepartment.substring(0, 2);
+    final otherDepartmentMaxCredit = courseScoreCredit.graduationInformation.outerDepartmentMaxCredit;
+
+    Map<String, List<CourseScoreInfoJson>> generalLesson = courseScoreCredit.getOtherDepartmentCourse(department);
```

Comment:
Suggest adding `final`.

Reply:
Addressed in [8332822](https://github.com/NEO-TAT/tat_flutter/pull/169/commits/8332822623e95cd30fe473ff8bab91450ac95224).